libffi-rs/src/high/mod.rs

```diff
@@ -78,6 +76,17 @@ pub use types::{CType, Type};
 pub mod call;
 pub use call::*;
 
+macro_rules! abort_on_panic {
+    ($msg:literal, $body:expr) => {
+        std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| { $body }))
+            .unwrap_or_else(|err| {
+                std::mem::forget(err); // defends against the issue that dropping `err` might panic
+                eprintln!($msg);
```

Comment:
This makes me wonder, should we be concerned about `eprintln!` panicking?

Maybe this should be done with a drop bomb rather than `catch_unwind`.

Reply:
`abort_on_panic` appears to use `writeln!`, and ignores its result: https://github.com/emk/abort_on_panic-rs/blob/master/src/lib.rs#L63-L64. I think it's best if we do something similar, otherwise a broken pipe error in `eprintln!` would probably panic and break things.

Reply:
Okay. I also switched to a drop-guard-based approach.